domain/externalcontroller/state/state.go

```diff
@@ -230,29 +272,31 @@ func (st *State) ModelsForController(
 		return nil, errors.Trace(err)
 	}
 
-	q := `
-SELECT uuid 
+	controller := Controller{
+		ID: controllerUUID,
+	}
+
+	stmt, err := st.Prepare(`
+SELECT &Model.uuid 
 FROM   external_model 
-WHERE  controller_uuid = ?`
+WHERE  controller_uuid = $Controller.uuid`, controller, Model{})
+	if err != nil {
+		return nil, errors.Annotate(err, "preparing select models for controller")
+	}
 
-	var modelUUIDs []string
-	err = db.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
-		rows, err := tx.QueryContext(ctx, q, controllerUUID)
+	var models []Model
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, stmt, controller).GetAll(&models)
 		if err != nil {
 			return errors.Trace(domain.CoerceError(err))
 		}
-		defer rows.Close()
-
-		for rows.Next() {
-			var modelUUID string
-			if err := rows.Scan(&modelUUID); err != nil {
-				return errors.Trace(err)
-			}
-			modelUUIDs = append(modelUUIDs, modelUUID)
-		}
 
 		return nil
 	})
 
+	modelUUIDs := transform.Slice[Model, string](
+		models,
+		func(m Model) string { return m.ID },
+	)
```

Comment:
This is unfortunate. I wonder if sqlair needs the inverse of a slice parameter.

```
SELECT &ModelUUIDs[:]
FROM   external_model 
WHERE  controller_uuid = $Controller.uuid
```

Reply:
This is something we were hoping would go away when we added scaler types (which was blocked). You could have just done `SELECT uuid AS &ModelUUID` and then used `GetAll`.

Reply:
Maybe a use case from the aspect of symmetry.